packages/terra-clinical-onset-picker/tests/nightwatch/component/Controlled.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import Base from 'terra-base';
+import moment from 'moment';
+import OnsetPicker from '../../../lib/OnsetPicker';
+
+const locale = document.getElementsByTagName('html')[0].getAttribute('lang');
+
+class Controlled extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // Lock date to September 20, 2016
+    moment.now = () => new Date(2016, 8, 20);
+
+    this.state = {
+      precision: 'before',
+      granularity: 'YEAR',
+      onsetDate: moment('2013-02-14'),
+    };
+
+    this.handleGranularity = this.handleGranularity.bind(this);
+    this.handlePrecision = this.handlePrecision.bind(this);
+    this.handleOnset = this.handleOnset.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleGranularity(granularity) {
+    this.setState({ granularity });
+  }
+
+  handlePrecision(precision) {
+    this.setState({ precision });
+  }
+
+  handleOnset(onsetDate) {
+    this.setState({ onsetDate });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    this.setState({
+      submittedData: {
+        precision: this.state.precision,
+        granularity: this.state.granularity,
+        onsetDate: this.state.onsetDate,
+      },
+    });
+  }
+
+  render() {
+    return (
+      <Base locale={locale}>
+        <p>Onset picker with onChange handlers, birthdate of 2011-09-20.</p>
+        <p>Current date is locked to September 20, 2016</p>
+        <p>Testing returns from handlers are correct, if they are the json object should get updated on changes.</p>
```

Comment:
Do you really need this extra file to test this? Your Less Month and Less year use the onChange handlers as well...

Reply:
I need an example with the handlers and an age of more than at least 2 years so I can test the month cap. So the lessMonth and lessYear wouldn't work out for this.